analyses/cms-open-data-ttbar/statistical_interference/agc_sample.py

```diff
@@ -0,0 +1,310 @@
+        """
+        channel_name = str(histoname.split("_")[0])
+
+        file = ROOT.TFile(histofile, "READ")
+        dir = file.GetDirectory(histopath)
+        hist_top = dir.Get(histoname)
+
+        hist_nominal_file = ROOT.TFile(self.GetInputFile(), "READ")
+        hist_nominal_name = self.GetHistoName()
+        hist_nominal_directory = hist_nominal_file.GetDirectory(
+            self.GetHistoPath()
+        )
+        hist_nominal = hist_nominal_directory.Get(hist_nominal_name)
+
+        norm_factor_up = hist_top.Integral() / hist_nominal.Integral()
+        h_new = hist_top.Clone(
+            channel_name
+            + "_"
+            + str(self.GetName())
+            + "_"
+            + name
+            + "_norm_plus_shape_up_clone"
+        )
+        h_new.Scale(1 / norm_factor_up)
+
+        h_down = hist_nominal.Clone(
+            channel_name
+            + "_"
+            + str(self.GetName())
+            + "_"
+            + name
+            + "_norm_plus_shape_down_clone"
+        )
+        h_down.Scale(2)
+        h_down.Add(h_new, -1)
+
+        output_file = ROOT.TFile(self.output_path, "UPDATE")
+
+        hist_up_name = str(
+            channel_name
+            + "_"
+            + str(self.GetName())
+            + "_"
+            + name
+            + "_norm_plus_shape_up"
+        )
+        hist_down_name = str(
+            channel_name
+            + "_"
+            + str(self.GetName())
+            + "_"
+            + name
+            + "_norm_plus_shape_down"
+        )
+
+        h_new.Write(hist_up_name)
+        h_down.Write(hist_down_name)
+
+        output_file.Close()
+
+        histSys = ROOT.RooStats.HistFactory.HistoSys()
+
+        histSys.SetName(name)
+        histSys.SetHistoPathHigh("")
+        histSys.SetHistoPathLow("")
+
+        histSys.SetHistoNameHigh(hist_up_name)
+        histSys.SetHistoNameLow(hist_down_name)
+
+        histSys.SetInputFileLow(self.output_path)
+        histSys.SetInputFileHigh(self.output_path)
+
+        overallSys = ROOT.RooStats.HistFactory.OverallSys()
+        overallSys.SetName(name)
+        overallSys.SetLow(2 - norm_factor_up)
+        overallSys.SetHigh(norm_factor_up)
+
+        self.GetHistoSysList().push_back(histSys)
+        self.GetOverallSysList().push_back(overallSys)
+
+    def NonSymmetrize_AddNormPlusShapeHistoSys(
+        self,
+        name,
+        histoname_up,
+        histofile_up,
+        histopath_up,
+        histoname_down,
+        histofile_down,
+        histopath_down,
+    ):
+        channel_name = str(histoname_up.split("_")[0])
+
+        file = ROOT.TFile(histofile_up, "READ")
+        dir = file.GetDirectory(histopath_up)
+        hist_top = dir.Get(histoname_up)
+
+        hist_nominal_file = ROOT.TFile(self.GetInputFile(), "READ")
+        hist_nominal_name = self.GetHistoName()
+        hist_nominal_directory = hist_nominal_file.GetDirectory(
+            self.GetHistoPath()
+        )
```

Comment:
In general, I think your code has too many line breaks. Up to 120 characters is okay!
```suggestion
        hist_nominal_directory = hist_nominal_file.GetDirectory(self.GetHistoPath())
```

We use [black](https://github.com/psf/black) to format the Python code in the ROOT repository, with the `--line-length` option:
```bash
black --line-length=120 *
```
Maybe you can do the same for this repository? At least the new code you added.